main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -766,40 +766,42 @@ object ConstraintGen {
 
         (resTpe, resEff)
 
-      case Expr.InvokeConstructor2(clazz, exps, jvar, evar, loc) =>
+      case Expr.InvokeConstructor2(clazz, exps, jvar, jevar, evar, loc) =>
         // Γ ⊢ eᵢ ... : τ₁ ...    Γ ⊢ ι ~ JvmConstructor(k, eᵢ ...)
         // --------------------------------------------------------
-        // Γ ⊢ new k(e₁ ...) : k
+        // Γ ⊢ new k(e₁ ...) : k \ JvmToEff[ι]
         val clazzTpe = Type.getFlixType(clazz)
         val (tpes, effs) = exps.map(visitExp).unzip
-        val baseEffs = BaseEffects.of(clazz, loc)
-        c.unifyType(jvar, Type.UnresolvedJvmType(Type.JvmMember.JvmConstructor(clazz, tpes), loc), loc) // unify constructor
-        c.unifyType(evar, Type.mkUnion(Type.IO :: baseEffs :: effs, loc), loc) // unify effects
+        c.unifyType(jvar, Type.UnresolvedJvmType(Type.JvmMember.JvmConstructor(clazz, tpes), loc), loc)
+        c.unifyType(jevar, Type.JvmToEff(jvar, loc), loc)
```

Comment:
Can you try it without jevar? I'm 76% sure it's unneeded.

Reply:
It immediately crashes

Reply:
a.uwaterloo.flix.util.InternalCompilerException: Unexpected type: 'JvmToEff(iterator)'.
	at ca.uwaterloo.flix.language.phase.unification.BoolFormula$.fromEffType(BoolFormula.scala:144)
	at ca.uwaterloo.flix.language.phase.unification.BoolFormula$.fromEffType(BoolFormula.scala:138)
	at ca.uwaterloo.flix.language.phase.unification.TypeMinimization$.minimizeBoolAlg(TypeMinimization.scala:106)
	at ca.uwaterloo.flix.language.phase.unification.TypeMinimization$.minimizeType(TypeMinimization.scala:35)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.resolveOne(ConstraintSolver.scala:265)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.tryResolve$1(ConstraintSolver.scala:246)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.resolveOneOf(ConstraintSolver.scala:256)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.resolve(ConstraintSolver.scala:222)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.visitSpec(ConstraintSolver.scala:104)
	at ca.uwaterloo.flix.language.phase.typer.ConstraintSolver$.visitDef(ConstraintSolver.scala:50)